### PR TITLE
[csharp-netcore] Better handling of oneOf/anyOf with duplicated data type

### PR DESCRIPTION
**Pull request by wing328**
- Better handling of oneOf/anyOf with duplicated data type
- add tests
- remove `skipOneOfAnyOfGetter` option as it's no longer needed

to fix https://github.com/OpenAPITools/openapi-generator/issues/14350

cc @OpenAPITools/generator-core-team 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/csharp-netcore/libraries/generichost/JsonConverter.mustache [hunk 1]
```diff
@@ -323,4 +329,4 @@
 
             writer.WriteEndObject();
         }
-    }
```
Comment (devhl-labs) [inline modules/openapi-generator/src/main/resources/csharp-netcore/libraries/generichost/JsonConverter.mustache:326]: Please remove this spacing change. Ill look at the other generichost changes later. Thanks for the work here.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/csharp-netcore/libraries/generichost/JsonConverter.mustache:326]: I want to. Never did I want to add this extra line but looks like that's a limitation in mustache.

Let me try other combination of mustache tags to see if I've any luck

Comment (wing328) [reply modules/openapi-generator/src/main/resources/csharp-netcore/libraries/generichost/JsonConverter.mustache]: Should be fixed via https://github.com/OpenAPITools/openapi-generator/pull/15377/commits/d8a11445a0f10abb6524de7dfe2f660d68947c48

**Comment by devhl-labs** [comment]
Here is a pr to this branch that will remove the extra spaces and also fixes a serialization issue.
https://github.com/OpenAPITools/openapi-generator/pull/15413

**Comment by wing328** [comment]
merged :)

**Pull request merged**